### PR TITLE
Design and implement cli for sending oracle script related transactions

### LINKED ISSUE #526: Design and implement CLI for sending oracle script related transactions

### PR DESCRIPTION
**Pull request by prin-r**
Fix #526

### CODE REVIEW

File: chain/x/zoracle/client/cli/tx.go [hunk 1]
```diff
@@ -285,3 +288,135 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
...
+	cmd.Flags().String(flagScript, "", "path to oracle script")
+	cmd.Flags().String(flagOwner, "", "owner of this oracle script")
+
+	return cmd
+}
+
+// GetCmdEditOracleScript implements the editing of oracle script command handler.
+func GetCmdEditOracleScript(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit-oracle-script (--id [id]) (--name [name]) (--script [path_to_script]) (--owner [owner])",
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Should we use positional argument for id?

File: chain/x/zoracle/client/cli/tx.go [hunk 2]
```diff
@@ -285,3 +288,135 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
 
 	return cmd
 }
+
+// GetCmdCreateOracleScript implements the create oracle script command handler.
+func GetCmdCreateOracleScript(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-oracle-script (--name [name]) (--script [path_to_script]) (--owner [owner])",
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: @taobun So are we going to use snake_case or camelCase for the things inside [...]?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/cli/tx.go (hunk 2 above)
Comment (sorawit) [reply chain/x/zoracle/client/cli/tx.go]: Let's go for dash-case. I think Cosmos-SDK usually uses that.

File: chain/x/zoracle/client/cli/tx.go [hunk 3]
```diff
@@ -285,3 +288,135 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
...
+		Short: "Create a new oracle script",
+		Args:  cobra.NoArgs,
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Create new oracle script that will be used by making a request.
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Create **a** new oracle script that will be used by data requests.

File: chain/x/zoracle/client/cli/tx.go [hunk 4]
```diff
@@ -285,3 +288,135 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
...
+}
+
+// GetCmdEditOracleScript implements the editing of oracle script command handler.
+func GetCmdEditOracleScript(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit-oracle-script (--id [id]) (--name [name]) (--script [path_to_script]) (--owner [owner])",
+		Short: "Edit an existing oracle script",
+		Args:  cobra.NoArgs,
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Edit an existing oracle script that will be used by making a request.
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: also fix here.

File: chain/x/zoracle/client/cli/tx.go (hunk 1 above)
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: ditto for dash-case

File: chain/x/zoracle/client/cli/tx.go [hunk 5]
```diff
@@ -285,3 +288,135 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
...
+			err = msg.ValidateBasic()
+			if err != nil {
+				return err
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	cmd.Flags().String(flagID, "", "ID of oracle script")
+	cmd.Flags().String(flagName, "", "name of oracle script")
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Please capitalize first words.

File: chain/x/zoracle/client/cli/tx.go [hunk 6]
```diff
@@ -285,3 +288,135 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
...
+
+			err = msg.ValidateBasic()
+			if err != nil {
+				return err
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	cmd.Flags().String(flagName, "", "name of oracle script")
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Please capitalize first words.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/cli/tx.go [hunk 7]
```diff
@@ -43,6 +43,8 @@ func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
 		GetCmdEditDataSource(cdc),
 		GetCmdRequest(cdc),
 		GetCmdReport(cdc),
+		GetCmdCreateOracleScript(cdc),
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go:44]: Move this to before Request and Report cmd.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — DISMISSED** [review]

File: chain/x/zoracle/client/cli/tx.go [hunk 8]
```diff
@@ -328,3 +330,133 @@ $ %s tx zoracle edit-data-source 1 --name coingecko-price --script ../price.sh -
...
+			if err != nil {
+				return err
+			}
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	cmd.Flags().String(flagName, "", "Name of oracle script")
+	cmd.Flags().String(flagScript, "", "Path to oracle script")
+	cmd.Flags().String(flagOwner, "", "Owner of this oracle script")
```
Comment (evilpeach) [inline chain/x/zoracle/client/cli/tx.go:459]: Make it consistent with others.

**Review by evilpeach — DISMISSED** [review]

File: chain/x/zoracle/client/cli/tx.go (hunk 8 above)
Comment (evilpeach) [reply chain/x/zoracle/client/cli/tx.go:459]: Ex. Name of this oracle script

**Review by evilpeach — APPROVED** [review]

**Review by sorawit — APPROVED** [review]

**Pull request merged**